### PR TITLE
[CC-25143] Contributor guide

### PR DESCRIPTION
**Pull request by erademacher**
Added a contributor guide section to DEVELOPMENT.md detailing how to run and add tests, perform a release, etc. Includes TF doc links, a general overview, and a brief Q&A with questions I've been asked recently.

Also updated the commit template with a checklist. I often forget to regenerate docs, so hopefully this will help. Someday, we'll have linters for that, but this'll work for now.

### CODE REVIEW

**Comment by erademacher** [comment]
Added some of the most recent contributors as reviewers. PTAL and let me known if there's anything else I should add!

**Review by jaylim-crl — APPROVED** [review]

**Comment by andy-kimball** [comment]
Nice, thanks for doing this.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -29,3 +29,95 @@ terraform provider
...
+
+## Suggested Reading
+
+- [Providers](https://developer.hashicorp.com/terraform/language/providers)
+- [Provider Framework](https://developer.hashicorp.com/terraform/plugin/framework)
+- [Debugging](https://developer.hashicorp.com/terraform/plugin/debugging)
+- [Testing](https://developer.hashicorp.com/terraform/plugin/testing)
+- [Best practice design principles](https://developer.hashicorp.com/terraform/plugin/best-practices/hashicorp-provider-design-principles)
+
+## Provider Philosiphy
```
Comment (jenngeorge) [inline DEVELOPMENT.md:51]: nit: "Philosophy"

**Review by jenngeorge — APPROVED** [review]
LGTM nice, thank you!!

**Comment by erademacher** [comment]
TFTRs!

**Pull request merged**